diff --git a/commerce_apd.module b/commerce_apd.module
index d6a504a..6c003dc 100644
--- a/commerce_apd.module
+++ b/commerce_apd.module
@@ -280,7 +280,9 @@ function commerce_apd_product_form_reference_submit(&$form, &$form_state) {
       // Loop through attached form
       foreach ($form2 as $field => $value) {
         // Fill the node form for submission.
-        $node_form[$field] = $form[$field];
+        if (isset($form[$field])) {
+          $node_form[$field] = $form[$field];
+        }
       }
 
       // Add other fields (if any) to the node class,
